tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+import os
+import re
+
+
+LOG_PATTERN = r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2},\d{3} \w{1,} \[\S{1,}\] \[\w{1,}.\w{2}:\d{1,}\] {}- .{1,}"
+
+
+def test_unrelated_logger_loaded_first_in_debug(run_python_code_in_subprocess):
+    """
+    When the tracer is imported after logging has been configured in debug mode,
+        the ddtrace logger does not override any custom logs settings.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+custom_logger = logging.getLogger('custom')
+custom_logger.setLevel(logging.WARN)
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+import ddtrace
+
+assert custom_logger.parent.name == 'root'
+assert custom_logger.level == logging.WARN
+
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err != b""
+    assert out == b""
+
+
+def test_unrelated_logger_loaded_last_in_debug(run_python_code_in_subprocess):
+    """
+    When the tracer is imported before logging has been configured in debug mode,
```

Comment:
the "in debug mode" shouldn't matter here right?

same for configuring this as logging to file or not?

if that is the case, we can probably parametrize this test case to vary the configuration options, but we still assert the same result.

For example:

```python
@pytest.mark.parametrize("dd_trace_debug", ["true", "false", None])
@pytext.mark.parametrize("dd_trace_log_file", [True, False])
def test_unrelated_logger_loaded_last(run_python_code_in_subprocess, tmpdir, dd_trace_debug, dd_trace_log_file):
    env = os.environ.copy()
    if dd_trace_debug is not None:
        env["DD_TRACE_DEBUG"] = dd_trace_debug
    if dd_trace_log_file:
        env["DD_TRACE_LOG_FILE"] = tmpdir + "/testlog.log"

    code = """
import ddtrace
import logging
custom_logger = logging.getLogger('custom')
custom_logger.setLevel(logging.WARN)
assert custom_logger.parent.name == 'root'
assert custom_logger.level == logging.WARN
"""

    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
    assert status == 0, err
    assert err != b""
    assert out == b""
```

With parametrize this way you will end up testing this test scenario with:

- DD_TRACE_DEBUG set to true and DD_TRACE_LOG_FILE set
- DD_TRACE_DEBUG set to true and DD_TRACE_LOG_FILE not set
- DD_TRACE_DEBUG set to false and DD_TRACE_LOG_FILE set
- DD_TRACE_DEBUG set to false and DD_TRACE_LOG_FILE not set
- DD_TRACE_DEBUG not set and DD_TRACE_LOG_FILE set
- DD_TRACE_DEBUG not set and DD_TRACE_LOG_FILE not set


which is quite exhaustive, but could be worth it. what do you think?

Reply:
Done, I added this to a few tests and it does seem useful to test several conditions with one test!